sirepo/pkcli/service.py

```diff
@@ -67,41 +72,69 @@ def _handle_signals(signums):
         finally:
             [signal.signal(x[0], x[1]) for x in o]
 
+    def _safe_kill_process(proc):
+        try:
+            proc.terminate()
+            proc.wait(1)
+        except (
+            ProcessLookupError,
+            ChildProcessError,
+            psutil.NoSuchProcess,
+        ):
+            pass
+        except (psutil.TimeoutExpired, subprocess.TimeoutExpired):
+            proc.kill()
+
     def _kill(*args):
         for p in processes:
             try:
-                p.terminate()
-                p.wait(1)
-            except (ProcessLookupError, ChildProcessError):
-                continue
-            except subprocess.TimeoutExpired:
-                p.kill()
-
-    def _start(service):
-        c = ["pyenv", "exec", "sirepo"]
-        c.extend(service)
+                for c in list(psutil.Process(p.pid).children(recursive=True)):
+                    _safe_kill_process(c)
+            except Exception:
+                # need to ignore exceptions while converting process children
+                # to a list so that the parent is still terminated
+                pass
+            _safe_kill_process(p)
+
+    def _start(
+        service, extra_environ=None, cwd=".", prefix=("pyenv", "exec", "sirepo")
+    ):
+        if extra_environ is not None:
```

Comment:
there will always be values.

Reply:
I believe this is solved, please check latest to make sure I understood the suggestions